db/multi_cf_iterator.h

```diff
@@ -65,52 +69,113 @@ class MultiCfIterator : public Iterator {
       assert(b.iterator->Valid());
       int c = comparator_->Compare(a.iterator->key(), b.iterator->key());
       assert(c != 0 || a.order != b.order);
-      return c == 0 ? a.order - b.order > 0 : c > 0;
+      return c == 0 ? a.order - b.order > 0 : CompareOp()(c, 0);
     }
 
    private:
     const Comparator* comparator_;
   };
-
   const Comparator* comparator_;
   using MultiCfMinHeap =
-      BinaryHeap<MultiCfIteratorInfo, MultiCfMinHeapItemComparator>;
-  MultiCfMinHeap min_heap_;
-  // TODO: MaxHeap for Reverse Iteration
+      BinaryHeap<MultiCfIteratorInfo,
+                 MultiCfHeapItemComparator<std::greater<int>>>;
+  using MultiCfMaxHeap = BinaryHeap<MultiCfIteratorInfo,
+                                    MultiCfHeapItemComparator<std::less<int>>>;
+
+  using MultiCfIterHeap = std::variant<MultiCfMinHeap, MultiCfMaxHeap>;
+
+  MultiCfIterHeap heap_;
+
+  enum Direction : uint8_t { kForward, kReverse };
+  Direction direction_ = kForward;
```

Comment:
If using variant instead of a union, I don't know what useful purpose `direction_` serves. It's just unnecessary maintenance of redundancy, with opportunity for consistency bugs, IMHO.

Reply:
I find it slightly more readable to follow the current direction, and it's more convenient than checking if the current heap is min or max.

I'll try to address this in a separate PR. I'm interested in seeing if we can make this cleaner without keeping `direction_` around